### PR TITLE
Update the project requirements for Ada cooks.

### PR DESCRIPTION
**Pull request by jnf**
New topics to cover in lecture:
- File uploads (probably a live code with CarrierWave)
- `has_and_belongs_to_many` (most groups are using this in Betsy, I believe)
- View helpers and/or Decorator pattern (maybe?).
- factory_girl set up and testing example

### CODE REVIEW

File: topic_resources/ada_cooks.md [hunk 1]
```diff
@@ -1,71 +1,88 @@
...
+  - Destroy an existing Recipe that I created
+    - Destroying a Recipe __does not__ destroy the Ingrdients or Gadgets, just the associations
+  - From a Recipe's detail page, I can save the Recipe to an existing Cookbook
+- __Ingredients__
+  - Create an Ingredient with...
+    - a unique `name` (required)
+    - an uploaded image of the Ingredient (optional)
+  - Edit the `name` and/or the uploaded image of an existing Ingredient that I created
+  - Destroy an existing Ingredient that I created
+- __Gadgets__
```
Comment (kariabancroft) [inline topic_resources/ada_cooks.md]: I think this is project is large enough without even considering gadgets. I think we could potentially remove this concept entirely and folks who finish earlier than others can work on additional styling and testing without needing to add any additional functionality.

File: topic_resources/ada_cooks.md [hunk 2]
```diff
@@ -1,71 +1,88 @@
...
+### As a Logged In User I can...
+- Log Out
+- __Cookbooks__
+  - See a list of Cookbooks I've already created, including...
+    - The number of Recipes associated with the Cookbook
+    - The number of unique Gadgets associated with the Cookbook
+    - The number of unique Recipes associated with the Cookbook
+  - See the details of a specific Cookbook, including...
+    - a list of links to the Recipes associated with the Cookbook
+    - a link to remove (unassociate) a Recipe from the Cookbook
```
Comment (kariabancroft) [inline topic_resources/ada_cooks.md]: Add a note that the Recipe will still exist, it has just removed the association

File: topic_resources/ada_cooks.md [hunk 3]
```diff
@@ -1,71 +1,88 @@
...
+    - a `preparation` (required)
+    - one or more Ingredients (required)
+    - zero or more Gadgets (optional)
+  - Edit an existing Recipe that I created by...
+    - Changing the `name`, `description`, uploaded image, and/or `preparation`
+    - Removing or Adding Ingredients
+    - Removing or Adding Gadgets
+  - Destroy an existing Recipe that I created
+    - Destroying a Recipe __does not__ destroy the Ingrdients or Gadgets, just the associations
+  - From a Recipe's detail page, I can save the Recipe to an existing Cookbook
```
Comment (kariabancroft) [inline topic_resources/ada_cooks.md]: Can this cookbook association then be edited?

File: topic_resources/ada_cooks.md [hunk 4]
```diff
@@ -1,71 +1,88 @@
...
+- __Ingredients__
+  - Create an Ingredient with...
+    - a unique `name` (required)
+    - an uploaded image of the Ingredient (optional)
+  - Edit the `name` and/or the uploaded image of an existing Ingredient that I created
+  - Destroy an existing Ingredient that I created
```
Comment (kariabancroft) [inline topic_resources/ada_cooks.md]: What happens with all of the associated Recipes? Do we want them to figure this out on their own?

File: topic_resources/ada_cooks.md (hunk 1 above)
Comment (jnf) [inline topic_resources/ada_cooks.md]: I'm totally fine with removing it completely. I think with the emphasis on testing, advanced AR stuffs, and introducing file handling, there's plenty of meat here.

File: topic_resources/ada_cooks.md (hunk 2 above)
Comment (jnf) [inline topic_resources/ada_cooks.md]: :+1:

File: topic_resources/ada_cooks.md (hunk 4 above)
Comment (jnf) [inline topic_resources/ada_cooks.md]: I've been really explicit with the other requirements, so I should probably be really explicit here too.

File: topic_resources/ada_cooks.md (hunk 3 above)
Comment (jnf) [inline topic_resources/ada_cooks.md]: Yes. I'll add that.

File: topic_resources/ada_cooks.md [hunk 5]
```diff
@@ -1,71 +1,75 @@
...
+    - Removing or Adding Ingredients
+  - Destroy an existing Recipe that I created
+    - Destroying a Recipe __does not__ destroy the Ingrdients, just the associations
+  - From a Recipe's detail page, I can save the Recipe to an existing Cookbook
+- __Ingredients__
+  - Create an Ingredient with...
+    - a unique `name` (required)
+    - an uploaded image of the Ingredient (optional)
+  - Edit the `name` and/or the uploaded image of an existing Ingredient that I created
+  - Destroy an existing Ingredient that I created (this does not destroy any associated Ingredients)
```
Comment (kariabancroft) [inline topic_resources/ada_cooks.md]: To my original comment - I guess I was wondering what would happen to Recipes that were also using this ingredient?

Comment (jnf) [inline topic_resources/ada_cooks.md]: Whoops.

**Comment by jnf** [comment]
JEREMY MERGE

**Pull request merged**